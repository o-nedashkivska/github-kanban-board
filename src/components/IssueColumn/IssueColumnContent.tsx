import { useSelector } from "react-redux";
import { Typography } from "antd";
import { SyncOutlined } from "@ant-design/icons";
import IssueCard from "../IssueCard";
import { getCurrentIssuesStatus } from "../../store/selectors";

import styles from "./issues-column.module.css";

interface IssueColumnContentProps {
  columnName: string;
  issues: Array<any>;
}
const { Title } = Typography;

const IssueColumnContent: React.FC<IssueColumnContentProps> = ({
  columnName,
  issues,
}) => {
  const status = useSelector(getCurrentIssuesStatus);

  if (status[columnName] === "loading")
    return <SyncOutlined className={styles.loader} spin />;

  return (
    <>
      {issues.length === 0 ? (
        <Title level={4}>No issues</Title>
      ) : (
        issues.map((issue, index) => (
          <IssueCard key={issue.id} index={index} {...issue} />
        ))
      )}
    </>
  );
};

export default IssueColumnContent;
