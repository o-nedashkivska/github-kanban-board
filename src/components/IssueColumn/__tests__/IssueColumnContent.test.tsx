import { screen } from "@testing-library/react";

import { renderWithProviders } from "../../../test-utils";

import IssueColumnContent from "../IssueColumnContent";

describe("IssueColumnContent", () => {
  describe("displays correct elements if status of column is loading", () => {
    const preloadedState = {
      issues: {
        status: { toDo: "loading", inProgress: "initial", done: "rejected" },
        limit: 0,
        allIssuesByRepo: {},
      },
    };

    test("displays spinner icon if status of column is loading", () => {
      renderWithProviders(
        <IssueColumnContent columnName="toDo" issues={[]} />,
        { preloadedState }
      );

      const spinner = screen.getByRole("img");

      expect(spinner).toBeInTheDocument();
      expect(spinner).toHaveAttribute("aria-label", "sync");
    });

    test("matches snapshot", () => {
      const { asFragment } = renderWithProviders(
        <IssueColumnContent columnName="toDo" issues={[]} />,
        { preloadedState }
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("displays correct elements if status of column is other than loading", () => {
    const issues = [
      {
        title: "Title 1",
        index: 0,
        number: 1011,
        comments: 11,
        created_at: new Date(),
      },
      {
        title: "Title 2",
        index: 1,
        number: 2022,
        comments: 22,
        created_at: new Date(),
      },
    ];

    test("displays correct text if issues array is empty", () => {
      const { asFragment } = renderWithProviders(
        <IssueColumnContent columnName="done" issues={[]} />
      );

      expect(screen.getByText("No issues")).toBeInTheDocument();

      expect(asFragment()).toMatchSnapshot();
    });

    test("displays correct issues", () => {
      const { asFragment } = renderWithProviders(
        <IssueColumnContent columnName="done" issues={issues} />
      );

      issues.forEach(({ title }) =>
        expect(screen.getByText(title)).toBeInTheDocument()
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
