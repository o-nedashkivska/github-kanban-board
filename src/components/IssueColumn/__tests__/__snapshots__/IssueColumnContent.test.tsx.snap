// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueColumnContent displays correct elements if status of column is loading matches snapshot 1`] = `
<DocumentFragment>
  <span
    aria-label="sync"
    class="anticon anticon-sync anticon-spin loader"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="sync"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M168 504.2c1-43.7 10-86.1 26.9-126 17.3-41 42.1-77.7 73.7-109.4S337 212.3 378 195c42.4-17.9 87.4-27 133.9-27s91.5 9.1 133.8 27A341.5 341.5 0 01755 268.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.7 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c0-6.7-7.7-10.5-12.9-6.3l-56.4 44.1C765.8 155.1 646.2 92 511.8 92 282.7 92 96.3 275.6 92 503.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8zm756 7.8h-60c-4.4 0-7.9 3.5-8 7.8-1 43.7-10 86.1-26.9 126-17.3 41-42.1 77.8-73.7 109.4A342.45 342.45 0 01512.1 856a342.24 342.24 0 01-243.2-100.8c-9.9-9.9-19.2-20.4-27.8-31.4l60.2-47a8 8 0 00-3-14.1l-175.7-43c-5-1.2-9.9 2.6-9.9 7.7l-.7 181c0 6.7 7.7 10.5 12.9 6.3l56.4-44.1C258.2 868.9 377.8 932 512.2 932c229.2 0 415.5-183.7 419.8-411.8a8 8 0 00-8-8.2z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`IssueColumnContent displays correct elements if status of column is other than loading displays correct issues 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered ant-card-small card css-dev-only-do-not-override-1kuana8"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-stretch ant-flex-vertical"
      >
        <span
          class="ant-typography css-dev-only-do-not-override-1kuana8"
        >
          Title 1
        </span>
        <span
          class="ant-typography ant-typography-secondary css-dev-only-do-not-override-1kuana8"
        >
          #1011 opened today
        </span>
        <span
          class="ant-typography ant-typography-secondary css-dev-only-do-not-override-1kuana8"
        >
          No assignee | Comments: 11
        </span>
      </div>
    </div>
  </div>
  <div
    class="ant-card ant-card-bordered ant-card-small card css-dev-only-do-not-override-1kuana8"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-stretch ant-flex-vertical"
      >
        <span
          class="ant-typography css-dev-only-do-not-override-1kuana8"
        >
          Title 2
        </span>
        <span
          class="ant-typography ant-typography-secondary css-dev-only-do-not-override-1kuana8"
        >
          #2022 opened today
        </span>
        <span
          class="ant-typography ant-typography-secondary css-dev-only-do-not-override-1kuana8"
        >
          No assignee | Comments: 22
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`IssueColumnContent displays correct elements if status of column is other than loading displays correct text if issues array is empty 1`] = `
<DocumentFragment>
  <h4
    class="ant-typography css-dev-only-do-not-override-1kuana8"
  >
    No issues
  </h4>
</DocumentFragment>
`;
